Sming/Libraries/ModbusMaster/samples/generic/app/application.cpp

```diff
@@ -0,0 +1,125 @@
+#include <SmingCore.h>
+#include <Debug.h>
+#include "application.h"
+#include <ModbusMaster.h>
+
+#define RS485_RE_PIN 15
+#define MB_SLAVE_ADDR 1
+#define SLAVE_REG_ADDR 1
+
+Timer mbLoopTimer;
+
+ModbusMaster mbMaster;
+HardwareSerial Serial1(UART1);
+
+uint16_t globalSeconds = 0;
+
+void mbLoop()
+{
+	globalSeconds++;
+
+	uint8_t numberOfRegistersToWrite = 1;
+	uint8_t bufferPosition = 0;
+	mbMaster.begin(MB_SLAVE_ADDR, Serial);
+	mbMaster.setTransmitBuffer(bufferPosition, globalSeconds);
+	mbMaster.writeMultipleRegisters(SLAVE_REG_ADDR, numberOfRegistersToWrite);
+
+	uint8_t nrOfRegistersToRead = 1;
+	uint8_t mbResult = mbMaster.readHoldingRegisters(SLAVE_REG_ADDR, nrOfRegistersToRead); //see also readInputRegisters
+	uint16_t result = 0;
+
+	if(mbResult == mbMaster.ku8MBSuccess) {
+		/*
+		for(i = 0; i < nrOfRegisters; i++) {
+			buffer[i] = mbMaster.getResponseBuffer(i);
+			Serial1.printf("Reg %d: %d\r\n", i, buffer[i]);
+		}
+		*/
+		debugf("Data from slave: %d", mbMaster.getResponseBuffer(0));
+	} else {
+		debugf("Res err: %d", result);
+	}
+
+	mbMaster.clearResponseBuffer();
+}
+
+void init()
+{
+	pinMode(RS485_RE_PIN, OUTPUT);
+	digitalWrite(RS485_RE_PIN, 0);
+	Serial.begin(115200, SERIAL_8N1, SERIAL_FULL);
```

Comment:
Perhaps this needs a separate definition, e.g. `COM_SPEED_MODBUS`?

Reply:
I did some other changes also, you may comment.